src/realm/object-store/impl/object_accessor_impl.hpp

```diff
@@ -320,9 +344,42 @@ inline util::Optional<UUID> CppContext::unbox(util::Any& v, CreatePolicy, ObjKey
 }
 
 template <>
-inline Mixed CppContext::unbox(util::Any&, CreatePolicy, ObjKey) const
+inline Mixed CppContext::unbox(util::Any& v, CreatePolicy, ObjKey) const
 {
-    throw std::logic_error("'Any' type is unsupported");
+    if (v.has_value()) {
+        if (v.type() == typeid(int)) {
```

Comment:
Each of these `v.type()` calls is a virtual function call that probably can't be optimized out so it's maybe worth calling it once and storing it in a local.

Reply:
Sure!